src/server/scripts/Commands/cs_misc.cpp

```diff
@@ -1356,6 +1357,45 @@ class misc_commandscript : public CommandScript
         return true;
     }
 
+    // kick player
+    static bool HandleKickDelayPlayerCommand(ChatHandler* handler, Optional<PlayerIdentifier> target, Optional<std::string_view> reason, std::string_view delay_time)
+    {
+        if (!target)
+        {
+            target = PlayerIdentifier::FromTarget(handler);
+        }
+
+        if (!target || !target->IsConnected())
+        {
+            return false;
+        }
+
+        auto targetPlayer = target->GetConnectedPlayer();
+
+        if (handler->GetSession() && target->GetGUID() == handler->GetSession()->GetPlayer()->GetGUID())
+        {
+            handler->SendSysMessage(LANG_COMMAND_KICKSELF);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        // check online security
+        if (handler->HasLowerSecurity(targetPlayer))
+        {
+            return false;
+        }
+
+        std::string kickReasonStr = handler->GetAcoreString(LANG_NO_REASON);
+        if (reason && !reason->empty())
+        {
+            kickReasonStr = std::string{ *reason };
+        }
+
+        targetPlayer->KickPlayer(kickReasonStr, Seconds(Acore::StringTo<int32>(delay_time).value_or(0)));
```

Comment:
I am not the best when it comes to CommandHandler, but cant we pass the delay_time argument directly as Chrono by just needing the user to specify ms/s/d when using the command, to avoid the 3 way conversion?

Reply:
Sadly I don't know. Every other command parsing it same way.

Reply:
Ok just checking